lib/Parse/Lexer.cpp

```diff
@@ -2302,42 +2302,34 @@ void Lexer::lexTrivia(syntax::TriviaList &Pieces, bool IsForTrailingTrivia) {
     if (IsForTrailingTrivia)
       break;
     NextToken.setAtStartOfLine(true);
-    LLVM_FALLTHROUGH;
-  case ' ':
-  case '\t':
-  case '\v':
-  case '\f': {
-    auto Char = CurPtr[-1];
-    // Consume consective same characters.
-    while (*CurPtr == Char)
-      ++CurPtr;
-
-    auto Length = CurPtr - TriviaStart;
-    switch (Char) {
-    case ' ':
-      Pieces.push_back(TriviaPiece::spaces(Length));
-      break;
+    switch (CurPtr[-1]) {
     case '\n':
-      // FIXME: CR+LF shoud form one trivia piece
-      Pieces.push_back(TriviaPiece::newlines(Length));
+      Pieces.appendOrSquash(TriviaPiece::newlines(1));
       break;
     case '\r':
-      Pieces.push_back(TriviaPiece::carriageReturns(Length));
-      break;
-    case '\t':
-      Pieces.push_back(TriviaPiece::tabs(Length));
-      break;
-    case '\v':
-      Pieces.push_back(TriviaPiece::verticalTabs(Length));
-      break;
-    case '\f':
-      Pieces.push_back(TriviaPiece::formfeeds(Length));
+      if (CurPtr[0] == '\n') {
+        Pieces.appendOrSquash(TriviaPiece::carriageReturnLineFeeds(1));
+        CurPtr++;
```

Comment:
Nitpick: We usually use `++CurPtr` for simple increment.

Reply:
ok. I will fix this.